framework/src/multiapps/MultiApp.C

```diff
@@ -1121,12 +1121,22 @@ rankConfig(dof_id_type rank,
 
   if (slot_for_rank[rank] < 0)
     // ranks assigned a negative slot don't have any apps running on them.
-    return {0, 0, 0, 0, false};
+    return {0, 0, 0, 0, false, 0};
   dof_id_type slot_num = slot_for_rank[rank];
 
   bool is_first_local_rank = rank == 0 || (slot_for_rank[rank - 1] != slot_for_rank[rank]);
   auto n_local_apps = apps_per_slot + 1 * (slot_num < leftover_apps);
 
+  processor_id_type my_first_rank = 0;
+  for (dof_id_type rankiter = rank; rankiter > 0; rankiter--)
```

Comment:
Your use of the correct id type (`processor_id_type`) highlights the use of the incorrect id type throughout the rest of this function. Could you change all the `dof_id_types` in this function to `processor_id_type`?

Reply:
Oh yes, nice catch, thanks Alex!